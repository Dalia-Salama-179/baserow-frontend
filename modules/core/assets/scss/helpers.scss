.clearfix::after {
  clear: both;
  content: '';
  display: table;
}

.hidden {
  display: none;
}

.visibility-hidden {
  visibility: hidden;
  transform: translateX(-10000px) translateY(-10000px);
}

.align-right {
  text-align: right;
}

.background-white {
  background-color: $white;
}

.color-primary {
  color: $color-primary-500 !important;
}

.color-success {
  color: $color-success-500 !important;
}

.color-warning {
  color: $color-warning-500 !important;
}

.color-error {
  color: $color-error-500 !important;
}

.margin-top-0 {
  margin-top: 0 !important;
}

.margin-top-1 {
  margin-top: 8px !important;
}

.margin-top-2 {
  margin-top: 16px !important;
}

.margin-top-3 {
  margin-top: 30px !important;
}

.margin-top-4 {
  margin-top: 40px !important;
}

.margin-bottom-0 {
  margin-bottom: 0 !important;
}

.margin-bottom-1 {
  margin-bottom: 8px !important;
}

.margin-bottom-2 {
  margin-bottom: 16px !important;
}

.margin-bottom-3 {
  margin-bottom: 30px !important;
}

.margin-bottom-4 {
  margin-bottom: 40px !important;
}

.margin-left-auto {
  margin-left: auto !important;
}

.margin-left-0 {
  margin-left: 0 !important;
}

.margin-left-1 {
  margin-left: 8px !important;
}

.margin-left-2 {
  margin-left: 16px !important;
}

.margin-right-auto {
  margin-right: auto !important;
}

.margin-right-0 {
  margin-right: 0 !important;
}

.margin-right-1 {
  margin-right: 8px !important;
}

.margin-right-2 {
  margin-right: 16px !important;
}

.resizing-horizontal {
  cursor: col-resize;
}

// This selector can only be used to temporarily show an element for calculating
// positions.
.forced-block {
  display: block !important;
}

// Sometimes we forcefully want to disable the the user-select property to initial
// because the element is for example being edited via a contenteditable attribute
// which does not work in combination with the user-select: none; property in Safari.
.forced-user-select-initial {
  user-select: initial !important;
}

.forced-text-overflow-initial {
  text-overflow: initial !important;
}

.prevent-scroll {
  overflow: hidden !important;
}

@keyframes spin {
  0% {
    transform: rotate(0);
  }

  50% {
    transform: rotate(360deg);
  }

  100% {
    transform: rotate(720deg);
  }
}
