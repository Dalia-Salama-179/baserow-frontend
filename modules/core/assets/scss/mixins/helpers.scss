@mixin fixed-height($height, $font-size) {
  height: $height;
  line-height: $height;
  font-size: $font-size;
}

@mixin center-text($width, $font-size, $height: $width) {
  @include fixed-height($height, $font-size);

  width: $width;
  text-align: center;
}

@mixin loading($size: 1.4rem) {
  position: relative;
  display: block;
  width: $size;
  height: $size;
  border-radius: 50%;
  border: 0.25em solid;
  border-color: $color-primary-500 transparent $color-primary-500 transparent;
  animation: spin infinite 1800ms;
  animation-timing-function: cubic-bezier(0.785, 0.135, 0.15, 0.86);
}

// The basically works the same as padding: 10px 0 0 10px or padding: 10px; only then
// resulting position: absolute; with corresponding top, right, bottom, left values.
// -------
// Example:
//  @include absolute(10px);
// Would result in:
//  position: absolute;
//  top: 10px; right: 10px; bottom: 10px; left: 10px;
// -------
// Example:
//  @include absolute(10px, 20px);
// Would result in:
//  position: absolute;
//  top: 10px; right: 20px; bottom: 10px; left: 20px;
// -------
// Example:
//  @include absolute(10px, 20px, 30px);
// Would result in:
//  position: absolute;
//  top: 10px; right: 20px; bottom: 30px;
@mixin absolute($top: null, $right: null, $bottom: null, $left: null) {
  position: absolute;

  @if ($top != null and $right == null and $bottom == null and $left == null) {
    top: $top;
    right: $top;
    bottom: $top;
    left: $top;
  }

  /* stylelint-disable-next-line at-rule-no-unknown */
  @else if ($top != null and $right != null and $bottom == null and
  $left == null) {
    top: $top;
    right: $right;
    bottom: $top;
    left: $right;
  }

  /* stylelint-disable-next-line at-rule-no-unknown */
  @else {
    @if $top != null {
      top: $top;
    }

    @if $right != null {
      right: $right;
    }

    @if $bottom != null {
      bottom: $bottom;
    }

    @if $left != null {
      left: $left;
    }
  }
}
