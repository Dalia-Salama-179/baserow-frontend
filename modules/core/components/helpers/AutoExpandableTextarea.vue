<template>
  <textarea
    ref="inputTextArea"
    :value="value"
    :placeholder="placeholder"
    class="input auto-expandable-textarea"
    :style="{
      height: textBoxSize + 'px',
      overflow: textBoxOverflow,
    }"
    rows="1"
    @input="$emit('input', $event.target.value)"
    @click="$emit('click', $event)"
    @keyup="$emit('keyup', $event)"
    @keydown="$emit('keydown', $event)"
    @blur="$emit('blur', $event)"
  />
</template>
<script>
/**
 * A blank textarea with no surrounding box styling (see AutoExpandableTextareaInput if
 * you want a nice styled box around this) which will automatically expand given the
 * users input to the maximum number of rows specified by the maxRows prop.
 */
export default {
  name: 'AutoExpandableTextarea',
  props: {
    value: {
      required: true,
      type: String,
    },
    loading: {
      required: false,
      default: false,
      type: Boolean,
    },
    placeholder: {
      required: false,
      type: String,
      default: '',
    },
    maxRows: {
      required: false,
      type: Number,
      default: 4,
    },
    startingRows: {
      required: false,
      type: Number,
      default: 1,
    },
  },
  data() {
    return {
      numTextAreaLines: this.startingRows,
    }
  },
  computed: {
    textBoxSize() {
      return 22 * Math.min(this.numTextAreaLines, this.maxRows)
    },
    textBoxOverflow() {
      return this.numTextAreaLines > this.maxRows ? 'auto' : 'hidden'
    },
  },
  watch: {
    value() {
      this.resizeTextArea()
    },
  },
  mounted() {
    this.resizeTextArea()
  },
  methods: {
    resizeTextArea() {
      this.$nextTick(() => {
        const inputTextArea = this.$refs.inputTextArea
        this.numTextAreaLines = this.calculateHeight(inputTextArea)
      })
    },
    /**
     * Taken from https://stackoverflow.com/questions/1760629/how-to-get-number-of-rows-in-textarea-using-javascript/1761203#1761203
     *
     * The key reason we need this is to resize a fully expanded textarea to something
     * smaller as a user deletes newlines or text. Hence we need to actually manipulate
     * the dom and lower the height of the textarea until we find overflow is occurring
     * again to re-find the correct min height.
     */
    calculateContentHeight(ta, scanAmount) {
      const origHeight = ta.style.height
      let height = ta.offsetHeight
      const scrollHeight = ta.scrollHeight
      const overflow = ta.style.overflow
      /// only bother if the ta is bigger than content
      if (height >= scrollHeight) {
        /// check that our browser supports changing dimension
        /// calculations mid-way through a function call...
        ta.style.height = height + scanAmount + 'px'
        /// because the scrollbar can cause calculation problems
        ta.style.overflow = 'hidden'
        /// by checking that scrollHeight has updated
        if (scrollHeight < ta.scrollHeight) {
          /// now try and scan the ta's height downwards
          /// until scrollHeight becomes larger than height
          while (ta.offsetHeight >= ta.scrollHeight) {
            ta.style.height = (height -= scanAmount) + 'px'
          }
          /// be more specific to get the exact height
          while (ta.offsetHeight < ta.scrollHeight) {
            ta.style.height = height++ + 'px'
          }
          height--
          /// reset the ta back to it's original height
          ta.style.height = origHeight
          /// put the overflow back
          ta.style.overflow = overflow
          return height
        } else {
          // We weren't able to resize this element inside this function call to figure
          // out it's true height so we give up and just return the scrollHeight.
          return scrollHeight
        }
      } else {
        return scrollHeight
      }
    },
    calculateHeight(ta) {
      const style = window.getComputedStyle
        ? window.getComputedStyle(ta)
        : ta.currentStyle

      // This will get the line-height only if it is set in the css,
      // otherwise it's "normal"
      const taLineHeight = parseInt(style.lineHeight, 10)
      // Get the scroll height of the textarea
      const taHeight = this.calculateContentHeight(ta, taLineHeight)
      // calculate the number of lines
      return Math.ceil(taHeight / taLineHeight)
    },
  },
}
</script>
