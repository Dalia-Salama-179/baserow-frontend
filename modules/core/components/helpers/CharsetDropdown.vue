<template>
  <Dropdown :value="value" :disabled="disabled" @input="$emit('input', $event)">
    <DropdownItem name="Unicode (UTF-8)" value="utf-8"></DropdownItem>
    <DropdownItem name="Arabic (ISO-8859-6)" value="iso-8859-6"></DropdownItem>
    <DropdownItem
      name="Arabic (Windows-1256)"
      value="windows-1256"
    ></DropdownItem>
    <DropdownItem name="Baltic (ISO-8859-4)" value="iso-8859-4"></DropdownItem>
    <DropdownItem
      name="Baltic (windows-1257)"
      value="windows-1257"
    ></DropdownItem>
    <DropdownItem
      name="Celtic (ISO-8859-14)"
      value="iso-8859-14"
    ></DropdownItem>
    <DropdownItem
      name="Central European (ISO-8859-2)"
      value="iso-8859-2"
    ></DropdownItem>
    <DropdownItem
      name="Central European (Windows-1250)"
      value="windows-1250"
    ></DropdownItem>
    <DropdownItem name="Chinese, Simplified (GBK)" value="gbk"></DropdownItem>
    <DropdownItem name="Chinese (GB18030)" value="gb18030"></DropdownItem>
    <DropdownItem name="Chinese Traditional (Big5)" value="big5"></DropdownItem>
    <DropdownItem name="Cyrillic (KOI8-R)" value="koi8-r"></DropdownItem>
    <DropdownItem name="Cyrillic (KOI8-U)" value="koi8-u"></DropdownItem>
    <DropdownItem
      name="Cyrillic (ISO-8859-5)"
      value="iso-8859-5"
    ></DropdownItem>
    <DropdownItem
      name="Cyrillic (Windows-1251)"
      value="windows-1251"
    ></DropdownItem>
    <DropdownItem
      name="Cyrillic Mac OS (x-mac-cyrillic)"
      value="x-mac-cyrillic"
    ></DropdownItem>
    <DropdownItem name="Greek (ISO-8859-7)" value="iso-8859-7"></DropdownItem>
    <DropdownItem
      name="Greek (Windows-1253)"
      value="windows-1253"
    ></DropdownItem>
    <DropdownItem name="Hebrew (ISO-8859-8)" value="iso-8859-8"></DropdownItem>
    <DropdownItem
      name="Hebrew (Windows-1255)"
      value="windows-1255"
    ></DropdownItem>
    <DropdownItem name="Japanese (EUC-JP)" value="euc-jp"></DropdownItem>
    <DropdownItem
      name="Japanese (ISO-2022-JP)"
      value="iso-2022-jp"
    ></DropdownItem>
    <DropdownItem name="Japanese (Shift-JIS)" value="shift-jis"></DropdownItem>
    <DropdownItem name="Korean (EUC-KR)" value="euc-kr"></DropdownItem>
    <DropdownItem name="Macintosh" value="macintosh"></DropdownItem>
    <DropdownItem
      name="Nordic (ISO-8859-10)"
      value="iso-8859-10"
    ></DropdownItem>
    <DropdownItem
      name="South-Eastern European (ISO-8859-16)"
      value="iso-8859-16"
    ></DropdownItem>
    <DropdownItem name="Thai (Windows-874)" value="windows-874"></DropdownItem>
    <DropdownItem
      name="Turkish (Windows-1254)"
      value="windows-1254"
    ></DropdownItem>
    <DropdownItem
      name="Vietnamese (Windows-1258)"
      value="windows-1258"
    ></DropdownItem>
    <DropdownItem
      name="Western European (ISO-8859-1)"
      value="iso-8859-1"
    ></DropdownItem>
    <DropdownItem
      name="Western European (Windows-1252)"
      value="windows-1252"
    ></DropdownItem>
    <DropdownItem name="Latin 3 (ISO-8859-3)" value="iso-8859-3"></DropdownItem>
  </Dropdown>
</template>

<script>
export default {
  // Please keep in sync with
  // src/baserow/contrib/database/api/export/serializers.py:SUPPORTED_EXPORT_CHARSETS
  name: 'CharsetDropdown',
  props: {
    value: {
      type: String,
      required: false,
      default: null,
    },
    disabled: {
      type: Boolean,
      required: false,
      default: false,
    },
  },
}
</script>
