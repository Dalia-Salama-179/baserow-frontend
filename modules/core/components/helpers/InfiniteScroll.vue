<template>
  <section
    ref="infiniteScroll"
    :class="{ 'infinite-scroll--reversed': reverse }"
    class="infinite-scroll"
    @scroll="handleScroll"
    v-on="$listeners"
  >
    <slot />
    <div
      v-show="currentCount < maxCount"
      ref="loadingWrapper"
      class="infinite-scroll__loading-wrapper"
    >
      <div v-if="loading" class="loading"></div>
    </div>
    <slot
      v-if="renderEnd && currentCount >= maxCount && wrapperHasScrollbar"
      name="end"
    >
      <div class="infinite-scroll__end-line"></div>
    </slot>
  </section>
</template>

<script>
/**
 * The InfiniteScroll component is designed to wrap a list paginated list of items.
 * Provide it with the currentCount of loaded items and the maxCount that is available
 * from the server and it will emit 'load-next-page' events when the user scrolls to
 * the end of the current items.
 *
 * If reverse is set then the list of items is rendered bottom to top and when the user
 * scrolls to the top the next page is loaded.
 *
 * Provide an end slot which will be shown when the user scrolls to the end of all
 * available items.
 */
export default {
  props: {
    currentCount: {
      type: Number,
      required: true,
    },
    maxCount: {
      type: Number,
      required: true,
    },
    pageSize: {
      type: Number,
      required: false,
      // Please keep in sync with the default page size in
      // backend/src/baserow/api/pagination.py
      default: 100,
    },
    loading: {
      type: Boolean,
      required: true,
    },
    reverse: {
      type: Boolean,
      required: false,
      default: false,
    },
    renderEnd: {
      type: Boolean,
      required: false,
      default: true,
    },
  },
  data() {
    return {
      // We don't want to show the end-line unless the user has scrolled somewhere and
      // hit the end. This also helps us properly align any end-line as it always has
      // a scrollbar and can be calculated as such.
      wrapperHasScrollbar: false,
    }
  },
  watch: {
    currentCount() {
      this.calculateIfWrapperHasScrollbar()
    },
  },
  created() {
    if (this.reverse) {
      // In reverse mode the start is actually the very bottom of the wrapper, so we
      // need to scroll there immediately.
      this.$nextTick(() => {
        this.scrollToStart()
      })
    }
    this.calculateIfWrapperHasScrollbar()
  },
  methods: {
    calculateIfWrapperHasScrollbar() {
      this.$nextTick(() => {
        const infiniteScroll = this.$refs.infiniteScroll
        this.wrapperHasScrollbar =
          infiniteScroll &&
          infiniteScroll.scrollHeight > infiniteScroll.clientHeight
      })
    },
    /**
     * Action listener called when the scroll wrapper scrolls, triggers a load next
     * page event if the user has scrolled to the end of the wrapper.
     */
    handleScroll({ target: { scrollTop, clientHeight, scrollHeight } }) {
      const height = clientHeight + this.$refs.loadingWrapper.clientHeight
      if (this.reverse) {
        if (-scrollTop + height >= scrollHeight) {
          this.loadNextPage()
        }
      } else if (scrollTop + height >= scrollHeight) {
        this.loadNextPage()
      }
    },
    /**
     * Emits a load-next-page event if there are still things to load in from the server
     * and we aren't already loading.
     */
    loadNextPage() {
      if (this.currentCount < this.maxCount && !this.loading) {
        const nextPage = Math.ceil(this.currentCount / this.pageSize) + 1
        this.$emit('load-next-page', nextPage)
      }
    },
    scrollToStart() {
      const infiniteScroll = this.$refs.infiniteScroll
      infiniteScroll.scrollTop = this.reverse ? 0 : infiniteScroll.scrollHeight
    },
  },
}
</script>
