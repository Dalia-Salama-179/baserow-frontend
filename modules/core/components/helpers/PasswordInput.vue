<!--
This helper component can only be used in a form which uses the password validation
from modules/core/validators.js
as ist expects the following vuelidate validation state params

- required
- maxLength
- minLength

In order to use this component in another component/form it needs a value
prop which can be passed via the 'v-model' directive. That means the state for the 
password value always lives in the parent component.
The parent component also needs to provide the vuelidate password validation in order
for this component to show all the possible validation errors.

Example Usage:

<PasswordInput
   v-model="PASSWORDSTATE"
   :validation-state="the vuelidate validation instance of the given password value"
/>

In case the password validation rules change, the passwordValidation object in the
validators file needs to be updated, as well as this component.

modules/core/validators.js

-->
<template>
  <div>
    <input
      :class="{ 'input--error': validationState.$error }"
      type="password"
      class="input input--large"
      :value="value"
      @blur="validationState.$touch()"
      @input="$emit('input', $event.target.value)"
    />
    <div
      v-if="validationState.$error && !validationState.required"
      class="error"
    >
      {{ $t('error.inputRequired') }}
    </div>
    <div
      v-if="validationState.$error && !validationState.maxLength"
      class="error"
    >
      {{
        $t('error.maxLength', {
          max: validationState.$params.maxLength.max,
        })
      }}
    </div>
    <div
      v-if="validationState.$error && !validationState.minLength"
      class="error"
    >
      {{
        $t('error.minLength', {
          min: validationState.$params.minLength.min,
        })
      }}
    </div>
  </div>
</template>

<script>
export default {
  name: 'PasswordInput',
  props: {
    validationState: {
      type: Object,
      required: true,
    },
    value: {
      type: String,
      required: true,
    },
  },
}
</script>
