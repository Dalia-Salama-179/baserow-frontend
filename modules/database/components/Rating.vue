<template functional>
  <div
    class="rating"
    :class="[
      data.staticClass,
      `color--${props.color}`,
      props.readOnly ? '' : 'editing',
    ]"
  >
    <i
      v-for="index in props.readOnly ? props.value : props.maxValue"
      :key="index"
      class="fas rating__star"
      :class="{
        [`fa-${props.ratingStyle}`]: true,
        'rating__star--selected': index <= props.value,
      }"
      @click="
        !props.readOnly &&
          listeners['update'] &&
          listeners['update'](index === props.value ? 0 : index)
      "
    />
  </div>
</template>

<script>
export default {
  name: 'Rating',
  props: {
    readOnly: {
      default: false,
      type: Boolean,
    },
    value: {
      required: true,
      validator: () => true,
    },
    maxValue: {
      required: true,
      type: Number,
    },
    ratingStyle: {
      default: 'star',
      type: String,
    },
    color: {
      default: 'dark-orange',
      type: String,
    },
  },
}
</script>
