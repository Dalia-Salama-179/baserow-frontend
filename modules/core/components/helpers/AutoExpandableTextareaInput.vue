<template>
  <div
    class="auto-expandable-textarea__container"
    :class="{
      'auto-expandable-textarea--loading-overlay': loading,
    }"
  >
    <div v-if="loading" class="auto-expandable-textarea--loading"></div>
    <AutoExpandableTextarea
      v-model="innerValue"
      :placeholder="placeholder"
      :max-rows="maxRows"
      :starting-rows="startingRows"
      @keydown.enter.exact.prevent="$emit('entered')"
    >
    </AutoExpandableTextarea>
  </div>
</template>
<script>
import AutoExpandableTextarea from '@baserow/modules/core/components/helpers/AutoExpandableTextarea'

/**
 * A textarea styled as an input box which will automatically expand given the users
 * input to the maximum number of rows specified by the maxRows prop. Prevents enter
 * by default and instead emits an 'entered' event for the parent to handle. Also shows
 * a loading overlay when the loading prop is set to true.
 */
export default {
  name: 'AutoExpandableTextareaInput',
  components: {
    AutoExpandableTextarea,
  },
  props: {
    value: {
      required: true,
      type: String,
    },
    loading: {
      required: false,
      default: false,
      type: Boolean,
    },
    placeholder: {
      required: false,
      type: String,
      default: '',
    },
    maxRows: {
      required: false,
      type: Number,
      default: 4,
    },
    startingRows: {
      required: false,
      type: Number,
      default: 1,
    },
  },
  computed: {
    innerValue: {
      get() {
        return this.value
      },
      set(value) {
        this.$emit('input', value)
      },
    },
  },
}
</script>
