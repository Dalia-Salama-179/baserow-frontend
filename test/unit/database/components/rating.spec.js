import { TestApp } from '@baserow/test/helpers/testApp'
import Rating from '@baserow/modules/database/components/Rating'

describe('Rating component', () => {
  let testApp = null

  beforeAll(() => {
    testApp = new TestApp()
  })

  afterEach(() => {
    testApp.afterEach()
  })

  const mountComponent = (
    props = { value: 3, maxValue: 5, readOnly: true },
    listeners = {}
  ) => {
    return testApp.mount(Rating, { propsData: props, listeners })
  }

  const changeValue = async (wrapper, value) => {
    const star = wrapper.find(`.rating :nth-child(${value})`)

    await star.trigger('click')
  }

  test('Default rating component', async () => {
    const wrapper = await mountComponent()
    expect(wrapper.element).toMatchSnapshot()
  })

  test('Customized rating component', async () => {
    const wrapper = await mountComponent({
      value: 3,
      maxValue: 5,
      readOnly: false,
      style: 'flag',
      color: 'dark-blue',
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('Test interactions with rating component', async () => {
    const onUpdate = jest.fn()
    const wrapper = await mountComponent(
      {
        value: 3,
        maxValue: 5,
        readOnly: false,
      },
      { update: onUpdate }
    )

    changeValue(wrapper, 1)
    expect(onUpdate).toHaveBeenCalledWith(1)

    changeValue(wrapper, 5)
    expect(onUpdate).toHaveBeenCalledWith(5)

    // If we click on current value, should set value to 0
    changeValue(wrapper, 3)
    expect(onUpdate).toHaveBeenCalledWith(0)
  })
})
