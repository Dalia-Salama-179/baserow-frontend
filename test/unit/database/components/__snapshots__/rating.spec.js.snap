// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating component Customized rating component 1`] = `
<div
  class="rating color--dark-blue editing"
>
  <i
    class="fas rating__star fa-star rating__star--selected"
  />
  <i
    class="fas rating__star fa-star rating__star--selected"
  />
  <i
    class="fas rating__star fa-star rating__star--selected"
  />
  <i
    class="fas rating__star fa-star"
  />
  <i
    class="fas rating__star fa-star"
  />
</div>
`;

exports[`Rating component Default rating component 1`] = `
<div
  class="rating color--dark-orange"
>
  <i
    class="fas rating__star fa-star rating__star--selected"
  />
  <i
    class="fas rating__star fa-star rating__star--selected"
  />
  <i
    class="fas rating__star fa-star rating__star--selected"
  />
</div>
`;
